static/app/views/onboarding/targetedOnboarding/components/createProjectsFooter.tsx

```diff
@@ -42,14 +44,29 @@ export default function CreateProjectsFooter({
   const {teams} = useTeams();
 
   const createProjects = async () => {
-    // TODO: add logic to prevent creating project if step repeated
     try {
       addLoadingMessage(t('Creating projects'));
+
+      const lastState: ClientState = await api.requestPromise(
+        `/organizations/${organization.slug}/client-state/onboarding/`
+      );
       const responses = await Promise.all(
-        platforms.map(platform =>
-          createProject(api, organization.slug, teams[0].slug, platform, platform)
-        )
+        platforms
+          .filter(platform => !(lastState.platforms && lastState.platforms[platform]))
```

Comment:
@Neo-Zhixing One edge case that might happen is that a project gets created but later deleted. In that case, we still would want to re-create the project.

Reply:
Ok so I thought about this for a bit, properly handling this could be rather involved.
1. Fetch a list of current projects.
2. Group the projects based on the platform
3. For each selected project, lookup its slug in client-state.
4. If it doesn't exist in client-state, create.
5. If it does exist in client state, see if there's a project with the same platform and slug.
6. If no, create. If yes, skip.

Instead of this, I recommend that we just show a "Deleted" state in setupDocs. Onboarding is supposed to be onboarding, it's not your full-feature admin panel.

Reply:
Also, what if the user change the platform of the project later on? I guess what I'm saying is that we don't really need to be too paranoid about the consistency here, as long as it's reasonable.

Reply:
@Neo-Zhixing I'm less concerned about the platform changing because even if that happens we'll show the instructions for the platform the user is asking to use during onboarding and they'll still be able to send a first event. But if it's deleted then things are gonna get pretty messed up if we try to load the project and ask the user to send an event.

IMO just filtering out deleted projects should be super easy. Just use the `withProjects` HoC, then add a filter step that filters out any platforms that have a stored project ID that can't be found in the list of projects. Can't we just do that?